libkmod/libkmod-index.c

```diff
@@ -689,7 +689,7 @@ static struct index_mm_node *index_mm_read_node(struct index_mm *idx, uint32_t o
 		first = read_char_mm(&p);
 		last = read_char_mm(&p);
 
-		if (first > last)
```

Comment:
makes sense, but then now we silently use a corrupted index - modprobe even returns success.

Reply:
Which is also true for `first > last` or out of memory conditions with failing memory allocations here.

If it's a subtle hint to add error checks of index_mm_read_node callers: Yes, we should do that. ;) Eventually we have to adjust the API though, because `index_dump` itself returns void.